aws_advanced_python_wrapper/simple_read_write_splitting_plugin.py

```diff
@@ -0,0 +1,303 @@
+            candidate_conn = None
+
+            try:
+                if connect_func is not None:
+                    candidate_conn = connect_func()
+                elif host_info is not None:
+                    candidate_conn = plugin_service_connect_func(host_info)
+                else:
+                    return None
+
+                if candidate_conn is None:
+                    self._delay()
+                    continue
+
+                actual_role = self._plugin_service.get_host_role(candidate_conn)
+
+                if actual_role != role:
+                    ReadWriteSplittingConnectionManager.close_connection(candidate_conn, self._plugin_service.driver_dialect)
+                    self._delay()
+                    continue
+
+                return candidate_conn
+
+            except Exception:
+                ReadWriteSplittingConnectionManager.close_connection(candidate_conn, self._plugin_service.driver_dialect)
+                self._delay()
+
+        return None
+
+    def old_reader_can_be_used(self, reader_host_info: HostInfo) -> bool:
+        # Assume that the old reader can always be used, no topology-based information to check.
+        return True
+
+    def need_connect_to_writer(self) -> bool:
+        # SetReadOnly(true) will always connect to the read_endpoint, and not the writer.
+        return False
+
+    def refresh_and_store_host_list(
+        self, current_conn: Optional[Connection], driver_dialect: DriverDialect
+    ):
+        # Endpoint based connections do not require a host list.
+        return
+
+    def should_update_writer_with_current_conn(
+        self, current_conn: Connection, current_host: HostInfo, writer_conn: Connection
+    ) -> bool:
+        return (
+            self.is_writer_host(current_host)
+            and current_conn != writer_conn
+            and (
+                not self._verify_new_connections
+                or self._plugin_service.get_host_role(current_conn) == HostRole.WRITER
+            )
+        )
+
+    def should_update_reader_with_current_conn(
+        self, current_conn: Connection, current_host: HostInfo, reader_conn: Connection
+    ) -> bool:
+        return (
+            self.is_reader_host(current_host)
+            and current_conn != reader_conn
+            and (
+                not self._verify_new_connections
+                or self._plugin_service.get_host_role(current_conn) == HostRole.READER
+            )
+        )
+
+    def is_writer_host(self, current_host: HostInfo) -> bool:
+        return (
+            current_host.host.casefold() == self._write_endpoint.casefold()
+            or current_host.url.casefold() == self._write_endpoint.casefold()
+        )
+
+    def is_reader_host(self, current_host: HostInfo) -> bool:
+        return (
+            current_host.host.casefold() == self._read_endpoint.casefold()
+            or current_host.url.casefold() == self._read_endpoint.casefold()
```

Comment:
I don't think we use `self._write_endpoint` and `self._read_endpoint` anywhere else. Could we perhaps just store the `casefold` version in `self._write_endpoint` and self._read_endpoint`? It's minor on performance, but doing self._write_endpoint.casefold() would make us do unnecessary string operations.